sources/dma_driver.c

```diff
@@ -0,0 +1,719 @@
+/**
+ * @file dma_driver.c
+ * 
+ * @author Sam Donnelly (samueldonnelly11@gmail.com)
+ * 
+ * @brief Direct Memory Access (DMA) functions 
+ * 
+ * @version 0.1
+ * @date 2022-10-27
+ * 
+ * @copyright Copyright (c) 2022
+ * 
+ */
+
+//================================================================================
+// Includes 
+
+#include "dma_driver.h"
+
+//================================================================================
+
+
+//================================================================================
+// Function Prototypes 
+
+/**
+ * @brief Stream status 
+ * 
+ * @details Reads the stream enable flag. This is used by the stream disable function 
+ *          to know when a stream is disabled (bit cleared). The enable flag can be cleared 
+ *          by hardware on the DMA end of transfer or if an error occurs. 
+ * 
+ * @param dma_stream : pointer to DMA port stream in question 
+ * @return uint8_t : stream status - 0 = disabled, 1 = enabled 
+ */
+uint8_t dma_stream_status(
+    DMA_Stream_TypeDef *dma_stream); 
+
+
+/**
+ * @brief DMA channel select 
+ * 
+ * @details Selects the channel for a given DMA stream. Each stream has up to 8 channels 
+ *          to choose from, each with a different function. Only one channel can be assigned 
+ *          to the stream. <br> 
+ *          Note: The channel can only be selected when the stream is disabled 
+ * 
+ * @see dma_channel_t 
+ * 
+ * @param dma_stream : pointer to DMA port stream being configured 
+ * @param channel : channel to assign to the stream 
+ */
+void dma_chsel(
```

Comment:
Nit:

Depending on if this is a conventional name for 'channel select', always consider using a more verbose function name to make it easier to understand.

Like 'dma_channelSelect()'